### PR TITLE
ref(issues): Use `SegmentedControl`

### PR DESCRIPTION
**Pull request by vuluongj20**
Use `SegmentedControl` instead of `ButtonBar` in Issue Details. See screenshots below.

https://github.com/getsentry/sentry/issues/43865

### CODE REVIEW

File: static/app/components/events/groupingInfo/groupingVariant.tsx [hunk 1]
```diff
@@ -238,14 +229,17 @@ class GroupVariant extends Component<Props, State> {
...
+        <SegmentedControl.Item key="relevant">
           {t('Contributing values')}
-        </Button>
-        <Button barId="all" size="xs" onClick={this.handleShowNonContributing}>
-          {t('All values')}
-        </Button>
-      </ContributingToggle>
+        </SegmentedControl.Item>
+        <SegmentedControl.Item key="all">{t('All values')}</SegmentedControl.Item>
+      </SegmentedControl>
```
Comment (vuluongj20) [inline static/app/components/events/groupingInfo/groupingVariant.tsx:242]: Screenshot 2023-02-03 at 11 02 08 AM

Screenshot 2023-02-03 at 11 01 46 AM

File: static/app/components/issueDiff/index.tsx [hunk 2]
```diff
@@ -130,14 +126,19 @@ class IssueDiff extends Component<Props, State> {
...
                 {t('Diff stack trace and message')}
-              </Button>
-              <Button barId="grouping" size="sm" onClick={this.toggleDiffMode}>
+              </SegmentedControl.Item>
+              <SegmentedControl.Item key="grouping">
                 {t('Diff grouping information')}
-              </Button>
-            </ButtonBar>
+              </SegmentedControl.Item>
+            </SegmentedControl>
```
Comment (vuluongj20) [inline static/app/components/issueDiff/index.tsx:141]: Screenshot 2023-02-03 at 10 46 43 AM
Screenshot 2023-02-03 at 10 47 08 AM

File: static/app/views/issueDetails/groupEventAttachments/groupEventAttachmentsFilter.tsx [hunk 3]
```diff
@@ -44,23 +45,38 @@ const GroupEventAttachmentsFilter = (props: Props) => {
...
+                <SegmentedControl.Item key="screenshot">
+                  {t('Screenshots')}
+                </SegmentedControl.Item>,
+              ]
+            : []),
+          <SegmentedControl.Item key="onlyCrash">
+            {t('Only Crash Reports')}
+          </SegmentedControl.Item>,
+        ]}
+      </SegmentedControl>
```
Comment (vuluongj20) [inline static/app/views/issueDetails/groupEventAttachments/groupEventAttachmentsFilter.tsx:79]: Screenshot 2023-02-02 at 5 19 55 PM

Screenshot 2023-02-02 at 5 19 46 PM

File: static/app/views/issueDetails/groupSimilarIssues/similarStackTrace/index.tsx [hunk 4]
```diff
@@ -159,14 +154,19 @@ function SimilarStackTrace({params, location, project}: Props) {
...
                 {t('Old Algorithm')}
-              </Button>
-              <Button barId="new" size="sm" onClick={toggleSimilarityVersion}>
+              </SegmentedControl.Item>
+              <SegmentedControl.Item key="new">
                 {t('New Algorithm')}
-              </Button>
-            </ButtonBar>
+              </SegmentedControl.Item>
+            </SegmentedControl>
```
Comment (vuluongj20) [inline static/app/views/issueDetails/groupSimilarIssues/similarStackTrace/index.tsx:169]: Screenshot 2023-02-02 at 5 09 31 PM

Screenshot 2023-02-02 at 5 09 19 PM

**Review by evanpurkhiser — APPROVED** [review]

File: static/app/views/issueDetails/groupEventAttachments/groupEventAttachmentsFilter.tsx [hunk 5]
```diff
@@ -44,23 +45,38 @@ const GroupEventAttachmentsFilter = (props: Props) => {
...
+            ? [
+                <SegmentedControl.Item key="screenshot">
+                  {t('Screenshots')}
+                </SegmentedControl.Item>,
+              ]
+            : []),
+          <SegmentedControl.Item key="onlyCrash">
+            {t('Only Crash Reports')}
+          </SegmentedControl.Item>,
+        ]}
```
Comment (evanpurkhiser) [inline static/app/views/issueDetails/groupEventAttachments/groupEventAttachmentsFilter.tsx:78]: this is contrived isn't it lol

```suggestion
          <SegmentedControl.Item key="all">{t('All Attachments')}</SegmentedControl.Item>
          {isMobilePlatform(project.platform) && (
	          <SegmentedControl.Item key="screenshot">
	            {t('Screenshots')}
	          </SegmentedControl.Item>
          }
          <SegmentedControl.Item key="onlyCrash">
            {t('Only Crash Reports')}
          </SegmentedControl.Item
```

Comment (vuluongj20) [reply static/app/views/issueDetails/groupEventAttachments/groupEventAttachmentsFilter.tsx:78]: It is! Unfortunately `react-aria` doesn't allow us to put `false` as a child inside `SegmentedControl`.  If we do it'll work fine, but there will be a TS error. I considered modifying `SegmentedControl`'s prop types to accept booleans, but worried there might be an error in the future (if `react-aria` suddenly changes the way they parse collection children). If you think it's fine, though, I'm down for doing that!

**Pull request merged**